src/components/GoalPreview/GoalPreview.tsx

```diff
@@ -32,40 +39,82 @@ const StyledModalHeader = styled(ModalHeader)`
 `;
 
 const StyledModalContent = styled(ModalContent)`
-    overflow: auto;
+    z-index: 2;
+`;
+
+const StyledStickyModalContent = styled(ModalContent)`
+    position: sticky;
+    top: 0;
+    height: fit-content;
 `;
 
 const StyledModalPreview = styled(ModalPreview)`
+    width: 850px;
     display: flex;
     flex-direction: column;
 `;
 
 const GoalPreviewModal: React.FC<GoalPreviewProps> = ({ shortId, goal, defaults, onClose, onDelete }) => {
+    const { setPreview } = useGoalPreview();
     const onPreviewClose = useCallback(() => {
         onClose?.();
     }, [onClose]);
 
-    const onEditMenuChange = useCallback((item: { onClick: () => void }) => {
-        item.onClick?.();
-    }, []);
+    const {
+        goalProjectChange,
+        onGoalStateChange,
+        goalTagsUpdate,
+        goalOwnerUpdate,
+        invalidate,
+        onGoalParticipantAdd,
+        onGoalParticipantRemove,
+    } = useGoalResource({ id: goal?.id }, { invalidate: { getById: shortId } });
+
+    const onGoalTagAdd = useCallback(
```

Comment:
Lets make issue about moving tag add/remove handlers to useGoalResource hook. They are fully copy and pasted here and on GoalPage.

Reply:
onGoalAssigneeChange is also a fully copy

https://github.com/taskany-inc/issues/issues/1853

Reply:
onGoalAssigneeChange removed here https://github.com/taskany-inc/issues/pull/1824